target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/common.c

```diff
@@ -1466,6 +1466,7 @@ static int __init rtl83xx_sw_probe(struct platform_device *pdev)
 	priv->ds->dev = dev;
 	priv->ds->priv = priv;
 	priv->ds->ops = &rtl83xx_switch_ops;
+	priv->ds->needs_standalone_vlan_filtering = true;
```

Comment:
this looks like an unrelated change, more importantly though; the commit message doesn't explain the why. Why do we need this. Only to configure vlan0? I read "somewhere" that vlan0 is 'special' and often avoided. Is that the reason? and where is the reference to vlan0 at all? is that just poor naming of the dsa_switch struct member?

Reply:
To be honest, I'm not exactly sure why this change is necessary. However, the log in 5.10 shows that vlan0 is being handled and seems to be necessary for the same behavior in 5.15.
If I removed this (```needs_standalone_vlan_filtering=false```), the communication between ethernet and switch was broken.